pyanaconda/modules/timezone/initialization.py

```diff
@@ -0,0 +1,136 @@
+#
+# Copyright (C) 2022  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import time
+import requests
+
+from pyanaconda.timezone import get_preferred_timezone, is_valid_timezone
+from pyanaconda.core.configuration.anaconda import conf
+from pyanaconda.core.constants import GEOLOC_DEFAULT_PROVIDER, GEOLOC_PROVIDER_FEDORA_GEOIP, \
+    GEOLOC_PROVIDER_HOSTIP, NETWORK_CONNECTION_TIMEOUT
+from pyanaconda.modules.common.constants.services import NETWORK
+from pyanaconda.modules.common.structures.timezone import GeolocationData
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.util import is_module_available
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class GeolocationTask(Task):
+    """Run geolocation"""
+    @property
+    def name(self):
+        return "Geolocate the system"
+
+    def run(self):
+        url = self._convert_provider_id_to_url(conf.timezone.geolocation_provider)
+
+        log.info("Geoloc: starting lookup using provider: %s (%s)",
+                 conf.timezone.geolocation_provider, url)
+        start_time = time.time()
+
+        if not self._wait_for_network():
+            log.error("Geoloc: no network connection")
+            return GeolocationData()
+
+        result = self._locate(url)
+        log.info("Geoloc: lookup finished in %1.1f seconds", time.time() - start_time)
+
+        if result.is_empty():
+            log.info("Geoloc: no result")
```

Comment:
This will be clear from some of the error logs in the `_locate` method. I would just log the result data here.